File: Periode 3/Java/Chapter 1/3. Constructors and Collections/Test/src/Node.java
/*
 * Van Jurrian moest dit weg
*/
public class Node
{
    public boolean IsEmpty;
    public int Value;
    public Node Tail;

    public Node(int value, Node tail)
    {
        this.IsEmpty = false;
        this.Value = value;
        this.Tail = tail;

    }
}

File: Periode 3/InClasspractice/LinkedList/Node.cs
namespace LinkedList
{

    public class NodeOrEmpty
    {
        public bool IsEmpty;
    }

    public class Empty : NodeOrEmpty
    {
        public Empty()
        {
            this.IsEmpty = true;
        }
    }

    public class Node : NodeOrEmpty
    {
        public int Value;
        public NodeOrEmpty Tail;

        public Node(int value, NodeOrEmpty tail)
        {
            this.IsEmpty = false;
            this.Value = value;
            this.Tail = tail;
        }
    }
}